File: leetcode/283.move-zeroes/main.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        left, right, length = 0, 0, len(nums)
        while right < length:
            if nums[right] != 0:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
            right += 1


if __name__ == "__main__":
    lst = [1, 0, 1, 0, 1, 1, 1, 0]
    Solution().moveZeroes(lst)
    print(lst)

File: leetcode/283.move-zeroes/main.php
<?php

class Solution
{
    function moveZeroes(&$nums)
    {
        $p = 0;
        for ($i = 0; $i < count($nums); $i++) {
            if ($nums[$i] != 0) {
                if ($p != $i) {
                    $tmp = $nums[$i];
                    $nums[$i] = $nums[$p];
                    $nums[$p] = $tmp;
                }
                $p++;
            }
        }
    }
}

$nums = [0, 1, 0, 2, 0, 0, -1];
(new Solution())->moveZeroes($nums);
print_r($nums);
